### PR TITLE
Use the 2nd network location changed as a trigger for movement

### PR DESCRIPTION
**Pull request by garvankeeley**
#1579

Based on the latest log in https://github.com/mozilla/MozStumbler/issues/1579#issuecomment-85991105

The first response from the network location changed is false, the location has not changed yet. Lets try the 2nd response.

### CODE REVIEW

File: libraries/stumbler/src/main/java/org/mozilla/mozstumbler/service/stumblerthread/motiondetection/MotionSensor.java [hunk 1]
```diff
@@ -138,10 +138,17 @@ public void stop() {
...
             public void onLocationChanged(Location location) {
-                AppGlobals.guiLogInfo("MotionSensor.NetworkLocationChangeDetector triggered.");
+                if (mLastLocation == null) {
+                    mLastLocation = location;
+                    return;
+                }
+
+                AppGlobals.guiLogInfo("MotionSensor.NetworkLocationChangeDetector triggered. (" +
+                    Math.round(location.distanceTo(mLastLocation)) + "m)");
                 Intent sendIntent = new Intent(ACTION_USER_MOTION_DETECTED);
```
Comment (crankycoder) [inline libraries/stumbler/src/main/java/org/mozilla/mozstumbler/service/stumblerthread/motiondetection/MotionSensor.java:160]: You need to set `mLastLocation` here as well or else you'll only set last location once.

Comment (garvankeeley) [inline libraries/stumbler/src/main/java/org/mozilla/mozstumbler/service/stumblerthread/motiondetection/MotionSensor.java:160]: I only want it set once after start(). I'll add a comment.

File: libraries/stumbler/src/main/java/org/mozilla/mozstumbler/service/stumblerthread/motiondetection/MotionSensor.java [hunk 2]
```diff
@@ -138,12 +138,31 @@ public void stop() {
...
+                    // Use the first location change to init the location. This means an initial movement
+                    // can be missed, and a subsequent movement is required. However this is the most reliable
+                    // means of initializing the location.
+                    mLastLocation = location;
+                    return;
+                }
+
+                final int distanceMeters = Math.round(location.distanceTo(mLastLocation));
+                if (distanceMeters < DIST_THRESHOLD_M) {
+                    // The threshold set in requestLocationUpdates is unreliable, I have seen false triggers, check here instead
```
Comment (crankycoder) [inline libraries/stumbler/src/main/java/org/mozilla/mozstumbler/service/stumblerthread/motiondetection/MotionSensor.java:155]: I believe this comment, but it frightens me.

File: libraries/stumbler/src/main/java/org/mozilla/mozstumbler/service/stumblerthread/motiondetection/MotionSensor.java [hunk 3]
```diff
@@ -138,12 +138,31 @@ public void stop() {
...
+                    return;
+                }
+
+                AppGlobals.guiLogInfo("MotionSensor.NetworkLocationChangeDetector triggered. (" + distanceMeters + "m)");
                 Intent sendIntent = new Intent(ACTION_USER_MOTION_DETECTED);
                 LocalBroadcastManager.getInstance(mContext).sendBroadcastSync(sendIntent);
+
+                // Under expected usage, after the motion detection broadcast, this class goes through a stop()/start() cycle,
```
Comment (crankycoder) [inline libraries/stumbler/src/main/java/org/mozilla/mozstumbler/service/stumblerthread/motiondetection/MotionSensor.java:163]: Ah right. I keep forgetting this.  Thanks for the comment.

**Pull request merged**